pypdf/annotations.py

```diff
@@ -25,54 +37,45 @@ def _get_bounding_rectangle(vertices: List[Tuple[float, float]]) -> RectangleObj
     return rect
 
 
-class AnnotationBuilder:
-    """
-    The AnnotationBuilder creates dictionaries representing PDF annotations.
+class AnnotationDictionary(DictionaryObject, ABC):
+    def __init__(self) -> None:
+        # "rect" should not be added here as PolyLine can automatically set it
+        self[NameObject("/Type")] = NameObject("/Annot")
 
-    Those dictionaries can be modified before they are added to a PdfWriter
-    instance via ``writer.add_annotation``.
 
-    See `adding PDF annotations <../user/adding-pdf-annotations.html>`_ for
-    it's usage combined with PdfWriter.
+class Text(AnnotationDictionary):
+    """
+    A text annotation.
+
+    Args:
+        rect: array of four integers ``[xLL, yLL, xUR, yUR]``
+            specifying the clickable rectangular area
+        text: The text that is added to the document
+        open:
+        flags:
     """
 
-    from ..types import FitType, ZoomArgType
-
-    @staticmethod
-    def text(
+    def __init__(
+        self,
+        *,
         rect: Union[RectangleObject, Tuple[float, float, float, float]],
         text: str,
         open: bool = False,
         flags: int = 0,
-    ) -> DictionaryObject:
-        """
-        Add text annotation.
+    ):
+        self[NameObject("/Subtype")] = NameObject("/Text")
+        self[NameObject("/Rect")] = RectangleObject(rect)
+        self[NameObject("/Contents")] = TextStringObject(text)
+        self[NameObject("/Open")] = BooleanObject(open)
+        self[NameObject("/Flags")] = NumberObject(flags)
```

Comment:
I would personnaly remove flags from the constructor parameters

Reply:
Why?

Reply:
I do understand that this parameter is optional but it will be presented in the hints. This parameter will be used only by a limited number of users I would have set-it/read-it through a property

Reply:
Ah, now I get it :bulb: You want to keep the constructor interface cleaner. I see the value there, but it also comes at the expense of discoverability.

Give me a few hours to finish other topics; I think I'll follow your suggestion :+1: 

Thank you :hugs:

Reply:
I just added it as a property and removed it from the constructor :-)